packages/cli/src/deploy/getLibraryMap.ts

```diff
@@ -0,0 +1,12 @@
+import { Hex } from "viem";
+import { Library } from "./common";
+
+export type LibraryMap = { [key: string]: Library & { address?: Hex } };
+
+export function getLibraryMap(libraries: readonly Library[]): LibraryMap {
+  return Object.fromEntries(libraries.map((library) => [getLibraryKey(library), library]));
```

Comment:
not blocking but I wonder if we should just return some interface with `get` and `set` and leave the internal representation and key shape as implementation details, then you can use this almost like a regular map but with 2-3 args for the key (path + name + maybe deployer address)